src/fmu/sumo/uploader/_upload_files.py

```diff
@@ -5,18 +5,103 @@
 """
 
 from concurrent.futures import ThreadPoolExecutor
+import json
+import yaml
+from fmu.dataio._utils import read_parameters_txt
+from fmu.dataio.dataio import ExportData
+from fmu.sumo.uploader._fileonjob import FileOnJob
+from fmu.sumo.uploader._logger import get_uploader_logger
 
 # pylint: disable=C0103 # allow non-snake case variable names
 
 
-def _upload_files(files, sumo_connection, sumo_parent_id, threads=4, sumo_mode="copy"):
+logger = get_uploader_logger()
+
+
+def maybe_add_parameters(
+    case_uuid,
+    realization_id,
+    parameters_path,
+    config_path,
+    sumo_connection,
+):
+    """Generate a parameters object from the parameters.txt file
```

Comment:
Should we note that it generates the object only if `parameters.txt does not already exist on Sumo`?

Reply:
Good idea, well spotted